### ISSUE TITLE
Drop support for jaeger-thrift

### ISSUE THREAD

**Issue by breedx-splk** [open-telemetry/opentelemetry-java#4577]
I'd like to propose that we drop support for the `jaeger-thrift` exporter. The [upstream repo](https://github.com/jaegertracing/jaeger-client-java) has been marked deprecated and they're actively encouraging folks to use OpenTelemetry instead. They're also refusing security patches. 

Furthermore, [the docs indicate](https://www.jaegertracing.io/docs/1.35/features/) 

```
Since v1.35, the Jaeger backend can receive trace data from the OpenTelemetry 
SDKs in their native OpenTelemetry Protocol (OTLP)].
```

Overall, I think this just makes sense and reduces an overwhelming number of choices for end users.
Also curious what @pavolloffay thinks.

**Comment by breedx-splk** [comment]
Related: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1973

**Comment by jkwatson** [comment]
We're currently using this at Verta, so I would prefer that we don't drop support for it quite yet.

**Comment by breedx-splk** [comment]
Yeah I'm proposing that we keep it in the splunk distro until we deprecate the Smart Agent, which won't happen until December, but I figured upstream could get ahead of that (or that we'd need this very discussion 😁 )

**Comment by jkwatson** [comment]
> Yeah I'm proposing that we keep it in the splunk distro until we deprecate the Smart Agent, which won't happen until December, but I figured upstream could get ahead of that (or that we'd need this very discussion 😁 )

👍🏽  It won't be hard for us to switch over to using grpc, rather than thrift for our jaeger exporter; I just need some lead time to make sure it gets on the schedule.

**Comment by pavolloffay** [comment]
No objections from the Jaeger community. 

We recommend people to use Jaeger gRPC and/or switch to OTLP.

**Comment by anuraaga** [comment]
@pavolloffay I have heard from some users that while gRPC is the standard for Jaeger collector, UDP+Thrift is still idiomatic for Jaeger agent so they try to use it. I guess the jaeger-agent at least doesn't document supporting gRPC

https://www.jaegertracing.io/docs/1.35/deployment/#agent

Is it safe to say that we don't support the Jaeger agent with OTel SDKs and they must switch to using the OTel collector as an agent?

(Note for context, our README mentions "Jaeger over Thrift Over HTTP", but while it may not have been intentional, we support UDP by accepting `ThriftSender`)

**Comment by pavolloffay** [comment]
>  I guess the jaeger-agent at least doesn't document supporting gRPC

Correct, Jaeger agent does not support gRPC nor thrift over HTTP (the thrift exporter in this repo https://github.com/open-telemetry/opentelemetry-java/tree/main/exporters/jaeger-thrift)

I didn't know a sender can be configured https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/jaeger-thrift/src/main/java/io/opentelemetry/exporter/jaeger/thrift/JaegerThriftSpanExporterBuilder.java#L26 
I don't know if people are using this API.


@breedx-splk what is the motivation to remove jaeger-thrift? Is there any technical reason or just drop it because Jaeger clients are deprecated?

**Comment by pavolloffay** [comment]
> (Maybe) Jaeger deprecating jaeger-agent officially

There is no plan to deprecate jaeger-agent. At least I haven't heard anything about it in the Jaeger community.

The jaeger-java client was released recently to fix some CVEs.

**Comment by anuraaga** [comment]
> it reduces the chances of that happening and improves the OpenTelemetry as the standard.

I agree with this :) So listed out steps that we should probably take to remove it from OpenTelemetry completely. I think we just need a volunteer to drive that (hint hint :P)

**Comment by yurishkuro** [comment]
Any objections if I transfer this issue into OTEL Spec repo?

**Comment by jack-berg** [comment]
Nope!
